python-avd/pyavd/_eos_designs/structured_config/underlay/port_channel_interfaces.py

```diff
@@ -151,6 +129,35 @@ def port_channel_interfaces(self: AvdStructuredConfigUnderlayProtocol) -> None:
         # WAN HA interface for direct connection
         self._set_direct_ha_port_channel_interface()
 
+    def _validate_l3_port_channels(self: AvdStructuredConfigUnderlayProtocol) -> None:
+        """
+        Perform validation on l3_port_channels.
+
+        Raises AristaAvdInvalidInputsError if the data model is invalid.
+        """
+        parent_names = self._l3_port_channels_with_subinterfaces
+        regular_names = set()
+
+        for l3_port_channel in self.shared_utils.node_config.l3_port_channels:
+            if "." in l3_port_channel.name:
+                # Sub-interface specific validations.
+                if l3_port_channel.member_interfaces:
+                    msg = f"L3 Port-Channel sub-interface '{l3_port_channel.name}' has 'member_interfaces' set. This is not a valid setting."
+                    raise AristaAvdInvalidInputsError(msg)
+                if l3_port_channel._get("mode"):
+                    # Implies 'mode' is set when not applicable for a sub-interface.
+                    msg = f"L3 Port-Channel sub-interface '{l3_port_channel.name}' has 'mode' set. This is not a valid setting."
+                    raise AristaAvdInvalidInputsError(msg)
+            else:
+                regular_names.add(l3_port_channel.name)
+
+        # We need to ensure that parent port-channel interfaces are also included explicitly within list of port-channel interfaces.
+        if missing_parents := parent_names.difference(regular_names):
+            msg = (
+                f"One or more L3 Port-Channels '{', '.join(natural_sort(missing_parents))}' need to be specified as they have sub-interfaces referencing them."
```

Comment:
Can we add a negative test for this as well.

Reply:
Already there `missing-parent-of-l3-port-channel-sub-interface`. Carl is just moving this code around.